src/GovukNotify/Client/BaseClient.cs

```diff
@@ -129,8 +146,12 @@ private void HandleHTTPErrors(HttpResponseMessage response, string errorResponse
         }
 
         public Tuple<string, string> ExtractServiceIdAndApiKey(string fromApiKey)
-        {
-            if (fromApiKey.Length < 74 || string.IsNullOrWhiteSpace(fromApiKey) || fromApiKey.Contains(" "))
+        {
+#if NET6_0_OR_GREATER
+            if (string.IsNullOrWhiteSpace(fromApiKey) || fromApiKey.Length < 74 || fromApiKey.Contains(' '))
+#else
+            if (string.IsNullOrWhiteSpace(fromApiKey) || fromApiKey.Length < 74 || fromApiKey.Contains(" "))
```

Comment:
In the original code the `.Length` property was accessed before checking for null, which lead to `NullReferenceException`s when an api key was not set.